src/bluehawk/processor/Processor.ts

```diff
@@ -23,16 +23,22 @@ export type Listener = (result: ParseResult) => void | Promise<void>;
 
 export class Processor {
   processors: CommandProcessors = {};
-  listeners: Listener[] = [];
+  listeners = new Set<Listener>();
 
   // Subscribe to processed file events
   subscribe(listener: Listener): void {
-    this.listeners.push(listener);
+    if (this.listeners.has(listener)) {
+      console.warn(`Skipping already registered listener: ${listener}`);
+      return;
+    }
+    this.listeners.add(listener);
   }
 
   // Publish a processed file
   async publish(result: ParseResult): Promise<void> {
-    const promises = this.listeners.map((listener) => listener(result));
+    const promises = Array.from(this.listeners.values()).map((listener) =>
+      listener(result)
+    );
     await Promise.all(promises);
```

Comment:
https://github.com/mongodb-university/Bluehawk/pull/40/files#r574981832

Reply:
I don't know if we want to swallow the error, but you're right, it shouldn't propagate above this layer. This is exactly where it needs to be reported to the listener implementer can fix it.

As for await, there's not much difference in this case. It would be like the same as returning the result of Promise.all().

Reply:
We should probably name listeners. It would let us provide nice diagnostics.

Reply:
>As for await...
Scratch that, you're right. We are meant to fire and forget here. Updated.